src/textual/css/_style_properties.py

```diff
@@ -794,6 +795,37 @@ def __set__(self, obj: StylesBase, names: str | tuple[str] | None = None):
                 obj.refresh(layout=True)
 
 
+class BooleanProperty:
+    """Descriptor for getting boolean properties"""
+
+    def __get__(
+        self, obj: StylesBase, objtype: type[StylesBase] | None = None
+    ) -> bool | None:
+        """Get a ``bool``.
+
+        Args:
+            obj (Styles): The ``Styles`` object.
+            objtype (type[Styles]): The ``Styles`` class.
+
+        Returns:
+            bool | None: The boolean value
+        """
+        return cast("bool | None", obj.get_rule("color_auto", None))
+
+    def __set__(self, obj: Styles, value: bool | None) -> None:
+        """Set the boolean value
+
+        Args:
+            obj (Styles): The ``Styles`` object
+            value (bool | None): The value to set.
+        """
+        _rich_traceback_omit = True
+        if value is None:
+            obj.clear_rule("color_auto")
+        else:
+            obj.set_rule("color_auto", value)
```

Comment:
I think "auto_color" reads a little better.

Reply:
it might read better, however my choice of `color_auto` was to collocate this property with its attached `color` as they are sister properties, so they are together when we debug for example:
![Screenshot from 2022-07-01 09-15-30](https://user-images.githubusercontent.com/722388/176854331-bd0fd97e-916e-418d-837b-db832fc2f645.png)
It would also group the "auto" properties with their "main" one if we add more "_resolved at runtime 'auto' values_" like this one later one.

But no worries, I'll rename it :slightly_smiling_face:

Reply:
No, you're right. We should follow the established convention. Ok, lets stick to `color_auto`